core/src/main/java/apoc/hashing/Fingerprinting.java

```diff
@@ -114,18 +116,12 @@ public String fingerprintGraph(@Name(value = "propertyExcludes", defaultValue =
                     () -> new TreeMap<>()
             ));
 
-            // step 2: build inverse map
-            Map<String, Long> nodeHashToId = idToNodeHash.entrySet().stream().collect(Collectors.toMap(
-                    Map.Entry::getValue,
-                    Map.Entry::getKey,
-                    (o, o2) -> {
-                        throw new RuntimeException();
-                    },
-                    () -> new TreeMap<>()
-            ));
+            // step 2: build inverse entrySet()
+            final Stream<Map.Entry<String, Long>> inverseHashToNode = idToNodeHash.entrySet().stream()
```

Comment:
non va bene perché stai cambiando l'ordinamento, `nodeHashToId` prima della tua modifica una Treemap che internamente ordina le chiavi in modo "naturale", quindi sostanzialmente l'ordine era definito in base al valore della fingerprint di ogni nodo:
https://www.baeldung.com/java-treemap

Reply:
Vedi il codice che ti ho mandato su slack

Reply:
Ok cambiato, 
l'unica cosa è che a differenza del codice che mi hai mandato il `messageDigest.update(hash.getBytes());` dello step 3 l'ho messo nell' `ids.forEach(..)`, 
altrimenti falliva l'assertion a riga 102 del `FingerprintTest`, 
perché non terrebbe conto della presenza di più nodi uguali [ `CREATE (:Person{name:'ABC'})` ] e ne considererebbe solo uno nel calcolo.
Credo sia giusto così?

Reply:
Oki